### ISSUE TITLE
[coe][distributed] daemon-master.jar

### ISSUE THREAD

**Issue by astefano** [into-cps/into-cps.github.io#11]
Hi,

i'd like to experiment with the distributed coe. The commands from the manual:

`java -jar daemon.jar -host <public-ip> -ip4`

and

`$ java -Dcoe.fmu.custom.factory=org.intocps.orchestration.coe.distribution.DistributedFmuFactory -cp coe.jar:daemon-master.jar org.intocps.orchestration.coe.CoeMain`

ask for `daemon.jar` and `daemon-master.jar` but i don't know where these jars are.

Also, if you could provide a complete example using one of the use cases (the tank for instance) would be so useful!

Thank you,
lacramioara

**Comment by astefano** [comment]
Hi, 

Here an update from my side. I have downloaded the distributed coe from into-cps (Window > Show download manager), the version is 0.0.2. What i do is:

#########################
#start dcoe in terminal 1
#########################

`astefanoaei@fortiss-n-019 MINGW64 ~/distrib_coe$ java -Dcoe.fmu.custom.factory=org.intocps.orchestration.coe.distribution.DistributedFmuFactory -jar dcoe-0.0.2-jar-with-dependencies.jar`
`COE(Command line options: [port]`

`Version: 0.1.10`
`Now running on port 8082`

###########################
#start daemon in terminal 2
###########################

`astefanoaei@fortiss-n-019 MINGW64 ~/distrib_coe$ java -jar daemon-0.0.2-jar-with-dependencies.jar`
`Remote COE Deamon Ready`

################################
#create session, init & sim in terminal 3
################################

`astefanoaei@fortiss-n-019 MINGW64 ~/distrib_coe/test$ curl http://localhost:8082/createSession`
 `{"sessionId":73885}`

`astefanoaei@fortiss-n-019 ~/distrib_coe/test$ sessionId=73885`

`astefanoaei@fortiss-n-019 ~/distrib_coe/test$ curl -H "Content-Type: application/json" --data @config.json http://localhost:8082/initialize/$sessionId`
`[{"status":"Initialized","sessionId":73885,"lastExecTime":0,"avaliableLogLevels":{"{tank2}.tank2":[],"{controller}.controller":[{"name":"logAll","description":null},{"name":"logError","description":null},{"name":"logFmiCall","description":null},{"name":"Protocol","description":null},{"name":"VdmOut","description":null},{"name":"VdmErr","description":null}],"{tank1}.tank1":[]}}]`

`astefanoaei@fortiss-n-019 ~/distrib_coe/test$ curl -H "Content-Type: application/json" --data @config-sim.json http://localhost:8082/simulate/$sessionId`

At this point it just blocks, and in terminal 2 (where the daemon runs) i get:
 
 `Exception in thread "Thread-1" java.lang.NullPointerException`
`        at org.crescendo.fmi.CrescendoFmu.SetReal(CrescendoFmu.java:503)`
`        at org.intocps.java.fmi.service.ProtocolDriver.run(ProtocolDriver.java:115)`
`        at java.lang.Thread.run(Unknown Source)`

Any clue about this?

Cheers,
lacramioara

**Comment by lausdahl** [comment]
how did you specify the FMU path?

It may be a connection issue, check that you can connect to the daemon server and possible start it like: `java -jar daemon-0.0.2-jar-with-dependencies.jar -ip4 -host 192.168.1.241`

COE should be started like you already did: 
```
java -Dcoe.fmu.custom.factory=org.intocps.orchestration.coe.distribution.DistributedFmuFactory -jar distributed-fmu-bundle/dcoe-0.0.2-jar-with-dependencies.jar
```

here is a init json:

```json
{
 "algorithm": {
   "size": 0.1,
   "type": "fixed-step"
 },
 "livestream": {
   "{t}.tInstance": [
     "level"
   ]
 },
 "parameters": {
   "{c}.cInstance.minlevel": 2,
   "{c}.cInstance.maxlevel": 4
 },
 "connections": {
   "{t}.tInstance.level": [
     "{c}.cInstance.level"
   ],
   "{c}.cInstance.valve": [
     "{t}.tInstance.valve"
   ]
 },
 "fmus": {
   "{t}": "file:\/\/\/\/Users\/kel\/Downloads\/single-tank.fmu",
   "{c}": "rmi://192.168.1.241/FMU#file:\/\/\/\/Users\/kel\/Downloads\/single-tank-controller.fmu"
 }
}
```

**Comment by lausdahl** [comment]
I just released version 0.0.6 which should provide better error reporting please try this version. If it does not work then re-open this issue. I'm closing it now.
